gcc/c/c-lang.c

```diff
@@ -60,6 +60,8 @@ run_c_tests (void)
 {
   /* Run selftests shared within the C family.  */
   c_family_tests ();
+  diagnostic_c_tests ();
+  opt_problem_cc_tests ();
```

Comment:
I suppose, these two should rather move into `gcc/c-family/c-common.c:selftest::c_family_tests`, to make (or rather: keep) them effective for both C and C++ front ends?  Also, should (eventually, need not be now) rename to `c_family_[...]` (or similar) to make clear their context, or split up into generic vs. c-family parts, or similar.  As the discussion has been started already, this needs to be sorted out in GCC upstream, once GCC again is in development stage 1 (2022-04, I suppose).  I'm able to help, as necessary.

Reply:
I've moved and renamed the tests. Thanks